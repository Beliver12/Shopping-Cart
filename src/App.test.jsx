import { describe, it, expect } from 'vitest';



import ShopPage from "./routes/shopPage-1";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";


describe("ShopPage component", () => {
  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<ShopPage />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();

    render(<ShopPage />);
    const button = screen.getByRole("button", { name: "Add" });

    await user.click(button);

    expect(screen.getByRole("paragraph").textContent).toMatch(/1/i);
  });
});