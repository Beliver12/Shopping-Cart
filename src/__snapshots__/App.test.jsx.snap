// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage component > renders magnificent monkeys 1`] = `
<div>
  <div
    id="header"
  >
    <div
      id="home"
    >
      <div>
        <a
          class="logo"
          target="_blank"
        />
        <a
          href="https://react.dev"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <h1>
        Shopping Cart Project
      </h1>
    </div>
    <a
      href="/shop-page-1"
    >
      <h1>
        Open Shop
      </h1>
    </a>
    <a
      href="/"
    >
      <h1>
        Home-Page
      </h1>
    </a>
    <a
      href="/about"
    >
      <h1>
        About-Us
      </h1>
    </a>
    <a
      href="/cart"
    >
      <div
        id="cart-box"
      >
        <div
          id="cart"
        />
        <p>
          0
        </p>
      </div>
    </a>
  </div>
  <div
    id="content"
  >
    <a
      href="/shop-page-1"
    >
      <button>
        <span>
          Shop-Now
        </span>
      </button>
    </a>
  </div>
</div>
`;

exports[`ShopPage component > renders magnificent monkeys 1`] = `
<div>
  <div
    id="header"
  >
    <div
      id="home"
    >
      <div>
        <a
          class="logo"
          target="_blank"
        />
        <a
          href="https://react.dev"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <h1>
        Shopping Cart Project
      </h1>
    </div>
    <a
      href="/shop-page-1"
    >
      <h1>
        Open Shop
      </h1>
    </a>
    <a
      href="/"
    >
      <h1>
        Home-Page
      </h1>
    </a>
    <a
      href="/about"
    >
      <h1>
        About-Us
      </h1>
    </a>
    <a
      href="/cart"
    >
      <div
        id="cart-box"
      >
        <div
          id="cart"
        />
        <p>
          0
        </p>
      </div>
    </a>
  </div>
  <div
    id="content"
  >
    <a
      href="/shop-page-1"
    >
      <button>
        <span>
          Shop-Now
        </span>
      </button>
    </a>
  </div>
</div>
`;
